packages/components/src/box-model-overlay/test/index.tsx

```diff
@@ -0,0 +1,186 @@
+				ref={ targetRef }
+			/>
+			<BoxModelOverlay
+				data-testid="box-model-overlay"
+				showValues={ DEFAULT_SHOW_VALUES }
+				targetRef={ targetRef }
+			/>
+		</>
+	);
+
+	const box = screen.getByTestId( 'box' );
+	const overlay = screen.getByTestId( 'box-model-overlay' );
+
+	act( () => {
+		expect( targetRef.current ).toBe( box );
+	} );
+
+	expect( overlay ).toBeVisible();
+} );
+
+it( 'allows to call update imperatively via ref', async () => {
+	const overlayRef = createRef< BoxModelOverlayHandle >();
+
+	render(
+		<BoxModelOverlay
+			data-testid="box-model-overlay"
+			showValues={ DEFAULT_SHOW_VALUES }
+			ref={ overlayRef }
+		>
+			<div data-testid="box" style={ { height: 300, width: 300 } } />
+		</BoxModelOverlay>
+	);
+
+	const overlay = screen.getByTestId( 'box-model-overlay' );
+
+	const promise = new Promise( ( resolve ) => {
+		const mutationObserver = new window.MutationObserver( resolve );
+
+		mutationObserver.observe( overlay, {
+			attributes: true,
+			attributeFilter: [ 'style' ],
+		} );
+	} );
+
+	overlayRef.current.update();
+
+	const entries = await promise;
+
+	expect( entries[ 0 ].attributeName ).toBe( 'style' );
+} );
+
+it( 'should react to style changes', async () => {
+	render(
+		<BoxModelOverlay
+			data-testid="box-model-overlay"
+			showValues={ DEFAULT_SHOW_VALUES }
+		>
+			<div data-testid="box" style={ { height: 300, width: 300 } } />
+		</BoxModelOverlay>
+	);
+
+	const box = screen.getByTestId( 'box' );
+	const overlay = screen.getByTestId( 'box-model-overlay' );
+
+	const promise = new Promise( ( resolve ) => {
+		const mutationObserver = new window.MutationObserver( resolve );
+
+		mutationObserver.observe( overlay, {
+			attributes: true,
+			attributeFilter: [ 'style' ],
+		} );
+	} );
+
+	box.style.padding = '20px';
+
+	const entries = await promise;
+
+	expect( entries[ 0 ].attributeName ).toBe( 'style' );
+} );
+
+it( 'renders the overlay to where Popover.Slot is', async () => {
+	render(
+		<SlotFillProvider>
+			<BoxModelOverlay
+				data-testid="box-model-overlay"
+				showValues={ DEFAULT_SHOW_VALUES }
+			>
+				<div data-testid="box" style={ { height: 300, width: 300 } } />
+			</BoxModelOverlay>
+
+			<div data-testid="slot">
+				{ /* @ts-ignore-error: The type for Popover is wrong here. */ }
+				<Popover.Slot />
+			</div>
+		</SlotFillProvider>
+	);
+
+	const box = screen.getByTestId( 'box' );
+	const overlay = screen.getByTestId( 'box-model-overlay' );
+	const slot = screen.getByTestId( 'slot' );
+
+	expect( overlay ).toBeVisible();
+
+	expect( slot.contains( overlay ) ).toBe( true );
+	expect( slot.contains( box ) ).toBe( false );
+} );
+
+it( 'should correctly unmount the component', async () => {
+	const { unmount } = render(
+		<BoxModelOverlay
+			data-testid="box-model-overlay"
+			showValues={ DEFAULT_SHOW_VALUES }
+		>
+			<div data-testid="box" style={ { height: 300, width: 300 } } />
+		</BoxModelOverlay>
+	);
+
+	unmount();
+} );
```

Comment:
Testing that it unmounts without any errors. It'd be better if there's an explicit assertion to make though.

Reply:
TBH, it's more like a sanity check 😅. I'm okay if we just remove this and add it back when there's a need.

Reply:
The main reason why I'm asking is because we don't usually test for component un-mounting, and so I was curious if there was a specific reason for it (and if we could learn something from your PR!).

> I'm okay if we just remove this and add it back when there's a need.

Let's do this then, if you don't mind! Thank you :)